### PR TITLE
Reduce and add randomization to retry delays

### PR DESCRIPTION
**Pull request by kewilli**
Reduce and add randomization to retry delays

### CODE REVIEW

File: Public/Src/Cache/ContentStore/Distributed/Stores/DistributedContentCopier.cs [hunk 1]
```diff
@@ -170,8 +170,12 @@ protected override Task<BoolResult> ShutdownCoreAsync(OperationContext context)
                         break;
                     }
 
-                    TimeSpan waitDelay = _retryIntervals[attemptCount];
-                    Tracer.Warning(operationContext, $"{AttemptTracePrefix(attemptCount)} All replicas {hashInfo.Locations.Count} failed. Retrying for hash {hashInfo.ContentHash.ToShortString()}...");
+                    long waitTicks = _retryIntervals[attemptCount].Ticks;
+
+                    // Randomize the wait delay to `[0.5 * delay, 1.5 * delay)`
+                    TimeSpan waitDelay = TimeSpan.FromTicks((long)(waitTicks / 2 + waitTicks * ThreadSafeRandom.Generator.NextDouble()));
```
Comment (dannyvv) [inline Public/Src/Cache/ContentStore/Distributed/Stores/DistributedContentCopier.cs]: >2 + waitTick [](start = 79, length = 12)

you might want to add parentehsis to make the operator precedence more explicit

File: Public/Src/Cache/ContentStore/Distributed/Stores/DistributedContentCopier.cs [hunk 2]
```diff
@@ -170,8 +170,12 @@ protected override Task<BoolResult> ShutdownCoreAsync(OperationContext context)
                         break;
                     }
 
-                    TimeSpan waitDelay = _retryIntervals[attemptCount];
-                    Tracer.Warning(operationContext, $"{AttemptTracePrefix(attemptCount)} All replicas {hashInfo.Locations.Count} failed. Retrying for hash {hashInfo.ContentHash.ToShortString()}...");
+                    long waitTicks = _retryIntervals[attemptCount].Ticks;
+
+                    // Randomize the wait delay to `[0.5 * delay, 1.5 * delay)`
+                    TimeSpan waitDelay = TimeSpan.FromTicks((long)((waitTicks / 2) + (waitTicks * ThreadSafeRandom.Generator.NextDouble())));
+
```
Comment (dcwuser) [inline Public/Src/Cache/ContentStore/Distributed/Stores/DistributedContentCopier.cs]: Minor mathy stuff: once you involve floating point (via NextDouble()), probably easiest, safest, and fastest to stay in floating point space. That is: use TimeSpan.FromSeconds as TimeSpan.TotalSeconds. Moving back and forth between integer and floating point is a (tiny) perf cost and, more importantly, often risks weird behaviors such as a failed cast or integer division producing unexpected results (like 1/2 = 0).

**Pull request merged**